lib/sql.ts

```diff
@@ -16,310 +14,252 @@ export interface SQLOptions {
+		const source = await this.query({type: 'prepare', data: statement});
+		if (!source) return null;
+		return new Statement(source, this);
 	}
-	createProcess(options: SQLOptions) {
-		const process: ProcessWrapper = new DatabaseWrapper(options);
-		this.processes.push(process);
-		return process;
+	exec(data: string): Promise<{changes: number}> {
+		return this.query({type: 'exec', data});
 	}
-	listen() {}
-	destroyProcess(process: DatabaseWrapper) {
-		void process.release();
-		this.processes.splice(this.processes.indexOf(process), 1);
+	async runFile(file: string) {
+		const contents = await FS(file).read();
+		return this.query({type: 'exec', data: contents});
 	}
-}
-
-export const PM = new SQLProcessManager(module);
-const Database = getModule();
-
-function crashlog(err: Error, query?: any) {
-	process.send!(`THROW\n@!!@${JSON.stringify([err.name, err.message, 'a SQL process', query])}\n${err.stack}`);
-}
-
-let database: sqlite.Database | null;
-if (!PM.isParentProcess) {
-	let statementNum = 0;
-	const statements: Map<number, sqlite.Statement> = new Map();
-	const transactions: Map<string, sqlite.Transaction> = new Map();
-	let statementTable: {[k: string]: sqlite.Statement} = {};
-
-	const getStatementTable = () => {
-		if (Object.keys(statementTable).length !== statements.size) {
-			statementTable = {};
-			for (const statement of statements.values()) {
-				statementTable[statement.source] = statement;
-			}
+	async query(data: DatabaseQuery) {
+		const result = await super.query(data);
+		if (result === SQLiteErrors.StatementNotFound) {
+			throw new Error("Prepare a statement before using another database function with SQLDatabase.prepare");
 		}
-		return statementTable;
-	};
-
-	const {file, extension} = process.env;
-	database = Database ? new Database(file!) : null;
-	if (extension && database) {
-		const {
-			functions,
-			transactions: storedTransactions,
-			statements: storedStatements,
-			onDatabaseStart,
-			// eslint-disable-next-line @typescript-eslint/no-var-requires
-		} = require(`../${extension}`);
-		if (functions) {
-			for (const k in functions) {
-				database.function(k, functions[k]);
-			}
-		}
-		if (storedTransactions) {
-			for (const t in storedTransactions) {
-				const transaction = database.transaction(storedTransactions[t]);
-				transactions.set(t, transaction);
-			}
-		}
-		if (storedStatements) {
-			for (const k in storedStatements) {
-				const statement = database.prepare(storedStatements[k]);
-				statements.set(statementNum++, statement); // we use statementNum here to track with the rest
-			}
-			process.send!(`STATEMENTS\n${JSON.stringify(Object.fromEntries(statements))}\n`);
-		}
-		if (onDatabaseStart) {
-			onDatabaseStart(database);
+		if (result === SQLiteErrors.DatabaseNotFound) {
+			throw new Error("SQLite database was queried, but Config.usesqlite is off");
 		}
+		return result;
 	}
-	database?.pragma(`foreign_keys=on`);
```

Comment:
does this line go anywhere? was it actually necessary?

Reply:
Don't think it was needed. Consumers can choose themselves.

Reply:
OK.